specs/sharding/beacon-chain.md

```diff
@@ -103,460 +101,342 @@ The following values are (non-configurable) constants used throughout the specif
-class ShardBlobReference(Container):
-    slot: Slot
-    shard: Shard
-    # Builder of the data
-    builder_index: BuilderIndex
-    # Proposer of the shard-blob
-    proposer_index: ValidatorIndex
-    # Blob hash-tree-root for slashing reference
-    body_root: Root
-```
+### KZG
 
-### `ShardProposerSlashing`
+#### `hash_to_field`
 
 ```python
-class ShardProposerSlashing(Container):
-    slot: Slot
-    shard: Shard
-    proposer_index: ValidatorIndex
-    builder_index_1: BuilderIndex
-    builder_index_2: BuilderIndex
-    body_root_1: Root
-    body_root_2: Root
-    signature_1: BLSSignature
-    signature_2: BLSSignature
+def hash_to_field(x: Container):
+    return int.from_bytes(hash_tree_root(x), "little") % MODULUS
 ```
 
-### `ShardWork`
+#### `compute_powers`
 
 ```python
-class ShardWork(Container):
-    # Upon confirmation the data is reduced to just the commitment.
-    status: Union[                                                   # See Shard Work Status enum
-              None,                                                  # SHARD_WORK_UNCONFIRMED
-              AttestedDataCommitment,                                # SHARD_WORK_CONFIRMED
-              List[PendingShardHeader, MAX_SHARD_HEADERS_PER_SHARD]  # SHARD_WORK_PENDING
-            ]
+def compute_powers(x: uint256, n: uint64):
+    current_power = 1
+    powers = []
+    for i in range(n):
+        powers.append(uint256(current_power))
+        current_power = current_power * int(x) % MODULUS
 ```
 
-## Helper functions
-
-### Misc
-
-#### `next_power_of_two`
+#### `verify_kzg_proof`
 
 ```python
-def next_power_of_two(x: int) -> int:
-    return 2 ** ((x - 1).bit_length())
-```
+def verify_kzg_proof(commitment: KZGCommitment, x: uint256, y: uint256, proof: KZGCommitment) -> List[uint256]:
+    zero_poly = G2_SETUP[1].add(G2_SETUP[0].mult(x).neg())
 
-#### `compute_previous_slot`
-
-```python
-def compute_previous_slot(slot: Slot) -> Slot:
-    if slot > 0:
-        return Slot(slot - 1)
-    else:
-        return Slot(0)
+    assert (
+        bls.Pairing(proof, zero_poly)
+        == bls.Pairing(commitment, G2_SETUP[-degree + 1])
+    )
 ```
 
-#### `compute_updated_sample_price`
+#### `verify_degree_proof`
 
 ```python
-def compute_updated_sample_price(prev_price: Gwei, samples_length: uint64, active_shards: uint64) -> Gwei:
-    adjustment_quotient = active_shards * SLOTS_PER_EPOCH * SAMPLE_PRICE_ADJUSTMENT_COEFFICIENT
-    if samples_length > TARGET_SAMPLES_PER_BLOB:
-        delta = max(1, prev_price * (samples_length - TARGET_SAMPLES_PER_BLOB) // TARGET_SAMPLES_PER_BLOB // adjustment_quotient)
-        return min(prev_price + delta, MAX_SAMPLE_PRICE)
-    else:
-        delta = max(1, prev_price * (TARGET_SAMPLES_PER_BLOB - samples_length) // TARGET_SAMPLES_PER_BLOB // adjustment_quotient)
-        return max(prev_price, MIN_SAMPLE_PRICE + delta) - delta
+def verify_degree_proof(commitment: KZGCommitment, degree: uint64, proof: KZGCommitment):
```

Comment:
Better for formatting for the degree (and maybe rename it `degree_cap`) to be the number that the polynomial degree must be less than (so a zero polynomial would satisfy a degree cap of 0). That way the variable can be always non-negative.